alica_engine/src/engine/AlicaContext.cpp

```diff
@@ -147,13 +147,34 @@ void AlicaContext::reloadConfig()
 
 AlicaCommunicationHandlers AlicaContext::getCommunicationHandlers()
 {
-    return AlicaCommunicationHandlers{[this](std::shared_ptr<SyncTalk> st) { _engine->editSyncModul().onSyncTalk(st); },
-            [this](std::shared_ptr<SyncReady> sr) { _engine->editSyncModul().onSyncReady(sr); },
-            [this](const AllocationAuthorityInfo& aai) { _engine->editAuth().handleIncomingAuthorityMessage(aai); },
-            [this](std::shared_ptr<PlanTreeInfo> st) { _engine->editTeamObserver().handlePlanTreeInfo(st); },
-            [this](const SolverResult& sr) { _engine->editResultStore().onSolverResult(sr); },
-            [this](const AgentQuery& pq) { _engine->getTeamManager().handleAgentQuery(pq); },
-            [this](const AgentAnnouncement& pa) { _engine->editTeamManager().handleAgentAnnouncement(pa); }};
+    return AlicaCommunicationHandlers{[this](std::shared_ptr<SyncTalk> st) {
+                                          if (_engine)
```

Comment:
I think its better to check if `_initialised` is true

Reply:
and also start the communication after the engine is initialised

Reply:
> I think its better to check if `_initialised` is true

Yep, make sense

> and also start the communication after the engine is started

Not sure how to make it, this change to avoid crashes while running inside analyzer

Reply:
I think you can just move the call to `_communicator->startCommunication()` in `AlicaContext::init` to after the call to `_engine->start()`. It should not cause any issues I think, but need to double check (I'm not sure why it was started before engine start in the first place)

Reply:
I believe we tried this before when dealing with the logging pr, and did not switch the order because we did not have time to debug the crashing tests

Reply:
Ok, this check is important for instrumentation and didn't make code worse but barely makes it better here, unfortunately. I believe we need plan resources for investigation

Reply:
I thought the crashes were fixed with the fixes in dummy communication. @bjoernschroeder can you confirm?